upgrade/1.2/scripts/upgrade/prerequisites.sh

```diff
@@ -211,6 +211,62 @@ else
     echo "====> ${state_name} has been completed"
 fi
 
+state_name="REMOVE_DUPLICATE_BMC_DNS_RECORDS"
+state_recorded=$(is_state_recorded "${state_name}" $(hostname))
+if [[ $state_recorded == "0" ]]; then
+    echo "====> ${state_name} ..."
+    {
+        # Find a pod that works.
+        # List names of all Running vault pods, grep for just the cray-vault-# pods, and try them in
+        # turn until one of them has the IPMI credentials.
+        USERNAME=""
+        IPMI_PASSWORD=""
+        VAULT_TOKEN=$(kubectl get secrets cray-vault-unseal-keys -n vault -o jsonpath={.data.vault-root} | base64 -d)
+        for VAULT_POD in $(kubectl get pods -n vault --field-selector status.phase=Running --no-headers \
+                            -o custom-columns=:.metadata.name | grep -E "^cray-vault-(0|[1-9][0-9]*)$") ; do
+            USERNAME=$(kubectl exec -it -n vault -c vault ${VAULT_POD} -- sh -c \
+                "export VAULT_ADDR=http://localhost:8200; export VAULT_TOKEN=`echo $VAULT_TOKEN`; \
+                vault kv get -format=json secret/hms-creds/$TARGET_MGMT_XNAME" |
+                jq -r '.data.Username')
+            # If we are not able to get the username, no need to try and get the password.
+            [[ -n ${USERNAME} ]] || continue
+            export IPMI_PASSWORD=$(kubectl exec -it -n vault -c vault ${VAULT_POD} -- sh -c \
+                "export VAULT_ADDR=http://localhost:8200; export VAULT_TOKEN=`echo $VAULT_TOKEN`; \
+                vault kv get -format=json secret/hms-creds/$TARGET_MGMT_XNAME" |
+                jq -r '.data.Password')
+            break
+        done
+        # Make sure we found a pod that worked
+        [[ -n ${USERNAME} ]]
+
+        # Install our pit-init RPM and pull in any dependencies it has.
+        zypper --plus-repo https://artifactory.algol60.net/artifactory/csm-rpms/hpe/stable/sle-15sp2/ --no-gpg-checks in -y pit-init
+        IPMI_PASSWORD=changeme
```

Comment:
Do we want to do line 244?  Seems that will negate wipe out the real password we got above.